sources/packages/api/src/database/data.model.service.ts

```diff
@@ -25,8 +24,13 @@ export class DataModelService<DataModel extends BaseModel>  {
     return this.repo.remove([object]);
   }
 
-  async findById(id: number) {
-    return this.repo.findByIds([id]);
+  async findById(id: number): Promise<DataModel> {
+    const result = await this.repo.findByIds([id]);
+    if (result?.length === 1) {
+      return result[0];
+    }
+
+    return null;
```

Comment:
do we need to give `Promise<DataModel|null>`, since we are returning null in line 33

Reply:
Yes, it makes sense. Code changed and comments added.